lib/src/widgets/atomic/editors.dart

```diff
@@ -193,6 +193,54 @@ class ColorEditor extends StatelessWidget {
 	);
 }
 
+/// A widget to edit a color, backed by [TimerBuilder].
+class TimerEditor extends StatelessWidget {
+	/// The view model for this color.
+	final TimerBuilder model;
+	/// A widget that modifies the given view model's color.
+	const TimerEditor(this.model);
+
+	@override
+	Widget build(BuildContext context) => ProviderConsumer<TimerBuilder>.value(
+		value: model,
+		builder: (model) => AlertDialog(
+			title: const Text("Start a timer"),
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+				children: [
+          SizedBox(
+            height: 50, 
+            width: double.infinity,
+            child: TextField(
+              onChanged: model.setName,
+              decoration: const InputDecoration(hintText: "Timer Name"),
+            ),
+          ),
+          SizedBox(
+            height: 50, 
+            width: double.infinity,
+            child: TextField(
+              onChanged: model.setTime,
+              decoration: const InputDecoration(hintText: "Number of Minutes"),
+              inputFormatters: [FilteringTextInputFormatter.allow(RegExp(r"^[0-9]*$"))],
+            ),
+          ),
```

Comment:
```suggestion
          SizedBox(
            height: 50, 
            width: double.infinity,
            child: TextField(
              onChanged: (value) => model.name = value,
              decoration: const InputDecoration(hintText: "Timer Name"),
            ),
          ),
          SizedBox(
            height: 50, 
            width: double.infinity,
            child: TextField(
              onChanged: model.duration.update,
              decoration: const InputDecoration(hintText: "Number of Minutes"),
              inputFormatters: [FilteringTextInputFormatter.allow(RegExp(r"\d"))],
            ),
          ),
```
Uses the `NumberBuilder` to allow us to delete the `setName` and `setTime` functions

Reply:
u sure that this regex is correct?